test/tests/functional/pbs_job_dependency.py

```diff
@@ -353,3 +353,182 @@ def test_dependency_on_multiple_finished_job(self):
 
         a = {ATTR_depend: 'beforenotok:' + j1 + ":" + j2}
         self.check_job(a, reject_msg, 'F')
+
+    def check_depend_delete_msg(self, pjid, cjid):
+        """
+        helper function to check ia message that the dependent job (cjid)
+        is deleted because of the parent job (pjid)
+        """
+        msg = cjid + ";Job deleted as result of dependency on job " + pjid
+        self.server.log_match(msg)
+
+    def test_job_end_deleting_chain_of_dependency(self):
+        """
+        Submit a chain of dependent jobs and see if one of the running jobs
+        ends, all the dependent jobs (and their dependent jobs)
+        are also deleted.
+        """
+
+        a = {'job_history_enable': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        job = Job()
+        job.set_sleep_time(10)
+        j1 = self.server.submit(job)
+
+        a = {ATTR_depend: "afternotok:" + j1}
+        job = Job(attrs=a)
+        j2 = self.server.submit(job)
+
+        a = {ATTR_depend: "afterok:" + j2}
+        job = Job(attrs=a)
+        j3 = self.server.submit(job)
+
+        a = {ATTR_depend: "afterok:" + j1}
+        job = Job(attrs=a)
+        j4 = self.server.submit(job)
+
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=j1, extend='x')
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=j2, extend='x',
+                           max_attempts=3)
+        self.check_depend_delete_msg(j1, j2)
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=j3, extend='x',
+                           max_attempts=3)
+        self.check_depend_delete_msg(j2, j3)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=j4, max_attempts=3)
+
+    def test_qdel_deleting_chain_of_dependency(self):
+        """
+        Submit a chain of dependent jobs and see if one of the running jobs
+        is deleted, all the dependent jobs (and their dependent jobs)
+        are also deleted.
```

Comment:
Say something like jobs without their dependencies released.  Since literally j4 has a dependency on j1, but it is released by the time the delete happens.